spark/src/test/scala/org/apache/spark/sql/delta/DeltaVacuumSuite.scala

```diff
@@ -578,6 +586,230 @@ class DeltaVacuumSuite
+        Row(s"${deltaLog.dataPath}/_underscore_col_=10/test2.txt", 300000L, false, 0L)
+      )
+
+      val inventory = spark.createDataFrame(spark.sparkContext.parallelize(data), inventorySchema)
+
+      gcTest(deltaLog, clock)(
+        CreateFile("file1.txt", commitToActionLog = true, Map("_underscore_col_" -> "10")),
+        CreateFile("file2.txt", commitToActionLog = false, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test.txt", true, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test2.txt", false, Map("_underscore_col_" -> "10")),
+        CheckFiles(Seq("file1.txt", "_underscore_col_=10")),
+        CheckFiles(Seq("file2.txt", "_underscore_col_=10")),
+        LogicallyDeleteFile("_underscore_col_=10/test.txt"),
+        AdvanceClock(defaultTombstoneInterval + 1000),
+        ExpectFailure(
+          GCByInventory(dryRun = false, expectedDf = Seq(tempDir), inventory = Some(inventory)),
+          classOf[SparkException],
+          Seq(s"inventory schema should contain ${expectedInventorySchema.treeString}")
+        )
+      )
+    }
+  }
+
+  test("run vacuum by using inventory dataframe") {
+    withEnvironment { (tempDir, clock) =>
+      val deltaLog = DeltaLog.forTable(spark, tempDir, clock)
+      val txn = deltaLog.startTransaction()
+      val schema = new StructType().add("_underscore_col_", IntegerType).add("n", IntegerType)
+      val metadata =
+        Metadata(schemaString = schema.json, partitionColumns = Seq("_underscore_col_"))
+      txn.commit(metadata :: Nil, DeltaOperations.CreateTable(metadata, isManaged = true))
+
+      // Define the schema for the DataFrame
+      val dataSchema = StructType(
+        Seq(
+          StructField("path", StringType),
+          StructField("length", LongType),
+          StructField("isDir", BooleanType),
+          StructField("modificationTime", LongType)
+        ))
+
+      // Create a Seq of Rows containing the data
+      val data = Seq(
+        Row(s"${deltaLog.dataPath}/file1.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/file2.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/_underscore_col_=10/test.txt", 300000L, false, 0L),
+        Row(s"${deltaLog.dataPath}/_underscore_col_=10/test2.txt", 300000L, false, 0L)
+      )
+
+      val inventory = spark.createDataFrame(spark.sparkContext.parallelize(data), dataSchema)
+
+      gcTest(deltaLog, clock)(
+        CreateFile("file1.txt", commitToActionLog = true, Map("_underscore_col_" -> "10")),
+        CreateFile("file2.txt", commitToActionLog = false, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test.txt", true, Map("_underscore_col_" -> "10")),
+        CreateFile("_underscore_col_=10/test2.txt", false, Map("_underscore_col_" -> "10")),
+        CheckFiles(Seq("file1.txt", "_underscore_col_=10")),
+        CheckFiles(Seq("file2.txt", "_underscore_col_=10")),
```

Comment:
These check for the existence of files, so we can check all of them in one go.
```suggestion
        CheckFiles(Seq(
          "file1.txt",
          "file2.txt",
          "_underscore_col_=10/test.txt",
          "_underscore_col_=10/test2.txt")),
```

Reply:
Not seeing this one processed? I see it further down in the test but not here.